lib/models/repository-states/present.js

```diff
@@ -440,11 +504,14 @@ export default class Present extends State {
     });
   }
 
-  getCurrentBranch() {
-    return this.cache.getOrSet(Keys.currentBranch, async () => {
-      const {name, isDetached} = await this.git().getCurrentBranch();
-      return isDetached ? Branch.createDetached(name) : new Branch(name);
-    });
+  async getCurrentBranch() {
+    const {branch} = await this.getStatusBundle();
+    if (branch.head === '(detached)') {
```

Comment:
Fun question: what happens if someone has a branch named `(detached)`? :grin:

If it's possible to distinguish those cases in the actual status output, maybe we could use a `DETACHED` Symbol exported from `what-the-status`?